src/common/PropChangeEvent.js

```diff
@@ -7,52 +7,41 @@ export class PropChangeEvent extends CustomEvent {
 		this.name = name;
 		this.prop = prop;
 	}
-}
-
-export default function definePropChangeEvent (Class, name) {
-	let eventName = `${name}change`;
-	let onEventName = `on${eventName}`;
-
-	let eventClassName = name[0].toUpperCase() + name.slice(1) + "ChangeEvent";
 
-	Props.add(Class, onEventName, {
-		type: Function,
-		typeOptions: {
-			arguments: ["event"],
-		},
-	});
+	/**
+	 * Get a subclass for a specific prop
+	 * @param {*} name
+	 */
+	static for (name) {
+		let eventClassName = name[0].toUpperCase() + name.slice(1) + "ChangeEvent";
+		let CustomPropChangeEvent = class extends PropChangeEvent {
+			constructor (type, options = {}) {
+				super(type, options);
+
+				this[name] = options[name];
+			}
 
-	let CustomPropChangeEvent = class extends PropChangeEvent {
-		constructor (type, options = {}) {
-			super(type, options);
+			static dispatchFrom (element) {
+				let event = new this(name + "change", {
+					[name]: element[name],
+				});
+				element.dispatchEvent(event);
+			}
+		};
+		Object.defineProperty(CustomPropChangeEvent, "name", { value: eventClassName });
+		return CustomPropChangeEvent;
+	}
+}
 
-			this[name] = options[name];
-		}
-	};
-	Object.defineProperty(CustomPropChangeEvent, "name", { value: eventClassName });
+export default function definePropChangeEvent (Class, name) {
```

Comment:
Looking at the implementation, I can't see why we need the `Class` parameter—it's not used anywhere. Does this helper can be used without a class at all?

Reply:
Good point, removed!